xbmc/input/InputCodingTable.h

```diff
@@ -30,6 +30,26 @@ class IInputCodingTable
   virtual int GetType() { return TYPE_WORD_LIST; }
 
   virtual ~IInputCodingTable() {}
+  /*! \brief Called for the active keyboard layout when it's loaded, stick any initialization here
+
+      This won't be needed for most implementations so we don't set it =0 but provide a default
+      implementation.
+
+      \return true on success, false on any fatal errors
+  */
+  virtual bool Initialize() { return true; }
+
+  /*! \brief Called for the active keyboard layout when it's unloaded, stick any cleanup here
+
+      This won't be needed for most implementations so we don't set it =0 but provide a default
+      implementation.
+  */
+  virtual void UnInitialize() {}
+
+  /*! \brief Can be overridden if initialization is expensive to avoid calling initialize more than needed
+      \return true if initialization has beeen done and was successful, false otherwise.
+  */
+  virtual bool IsInitialized() const { return false; }
```

Comment:
I move the responsibility of tracking the "is initialized" state to the owner's class. This has two advantages:

* lowers the complexity of all derivatives of `IInputCodingTable` by a function and a state variable
* reduces code by moving the physical "is initialized" state variable into a contractual obligation for `Initialize()` (that it returns true if it's initialized and false if it's not initialized)

so the owning class would look like 

```c++
class OwnerOfInputTable
{
public:
  bool Initialize()
  {
    if (m_table->Initialize())
    {
      // Coding table is initialized and can be used
      return true;
    }
    // Coding table is not initialized
    return false;
  }

private:
  IInputCodingTable* m_table;
};
```

notice that the owning class following this pattern as well. You might have to cascade this change to other areas of the code, but each change equals lower complexity and less code.